test/fields.uts

```diff
@@ -751,6 +751,25 @@ assert p.data[3].data == 0xc102
 assert isinstance(p.payload, conf.raw_layer)
 assert p.payload.load == b'toto'
 
+= Test nested PacketListFields
+~ field
+# https://github.com/secdev/scapy/issues/3894
+
+class GuessPayload(Packet):
+     @classmethod
+     def dispatch_hook(cls, *args, **kargs):
+         return TestNestedPLF
+
+class TestNestedPLF(Packet):
+     fields_desc = [
+         ByteField('b', 0),
+         PacketListField('pl', [], GuessPayload)
+     ]
+
+p = TestNestedPLF(b'\x01' * 1000)
+assert len(p[Raw]) == 995
```

Comment:
why 995?

Reply:
The first `TestNestedPLF` instance isn't nested and it "consumes" the first byte. The next four `TestNestedPLF` instances are included in PacketListFields and consume four more bytes. The last 995 bytes go to `Raw`. `p` looks like
```python
###[ TestNestedPLF ]###
  b         = 1
  \pl        \
   |###[ TestNestedPLF ]###
   |  b         = 1
   |  \pl        \
   |   |###[ TestNestedPLF ]###
   |   |  b         = 1
   |   |  \pl        \
   |   |   |###[ TestNestedPLF ]###
   |   |   |  b         = 1
   |   |   |  \pl        \
   |   |   |   |###[ TestNestedPLF ]###
   |   |   |   |  b         = 1
   |   |   |   |  \pl        \
   |   |   |   |   |###[ Raw ]###
   |   |   |   |   |  load      = ...
```